core/core-util/src/main/java/ai/wanaku/core/uri/URIHelper.java

```diff
@@ -66,4 +70,12 @@ public static String buildUri(String scheme, String path, Map<String, ?> paramet
         uri.append(path);
         return buildFromBaseUri(uri, parameters);
     }
+
+    public static String addQueryParameters(String uri, Map<String, ?> parameters) {
+        boolean first = true;
+        if (uri.contains("?")) {first = false;}
```

Comment:
Small nitpick: make sure to use multilines for the if-else here. It makes it easier to read the code.